src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -213,8 +212,8 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
             }
             "type_name" => {
                 let tp_ty = substs.type_at(0);
-                let ty_name = LocalInternedString::intern(&tp_ty.to_string());
-                self.const_str_slice(ty_name)
+                let ty_name = rustc_mir::interpret::type_name(self.tcx, tp_ty);
+                OperandRef::from_const(self, ty_name).immediate_or_packed_pair(self)
```

Comment:
@oli-obk 	I assume Miri needs the same change then for its implementation of the intrinsic? Can you submit a PR?

Reply:
I submitted a [PR](https://github.com/rust-lang/miri/pull/761) for this.

Reply:
Uh, we can also just wait for https://github.com/rust-lang/rust/pull/61399 to get merged, then miri will be able to just remove its implementation